### PR TITLE
Adbscan fix

### PR DESCRIPTION
**Pull request by ljwolf**
This makes 
1. geopandas a function-level import 
2.  `adbscan` (formally) a scikit-learn estimator by making its inheritance from `_ClusterMixin` and `BaseEstimator` explicit. 

I don't think this affects anything, but it'd be good to check more than the existing test suite...

### CODE REVIEW

File: esda/adbscan.py [hunk 1]
```diff
@@ -500,7 +500,11 @@ def get_cluster_boundary(labels, xys, xy=["X", "Y"], n_jobs=1, crs=None, step=1)
     >>> polys[0].wkt
     'POLYGON ((0.7217553174317995 0.8192869956700687, 0.7605307121989587 0.9086488808086682, 0.9177741225129434 0.8568503024577332, 0.8126209616521135 0.6262871483113925, 0.6125260668293881 0.5475861559192435, 0.5425443680112613 0.7546476915298572, 0.7217553174317995 0.8192869956700687))'
     """
-
+    try:
+        from geopandas import GeoSeries
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError('geopandas is required in order '
+                                  ' to get cluster boundaries')
```
Comment (darribas) [inline esda/adbscan.py:507]: This looks good. The only thing I'd raise is whether we want to a) raise an error and stop if no `geopandas` or b) simply raise a warning and inform that the output `polys` will be returned as a list/ndarray/Series of geometries.

If b), then on L.523 we should `try` what's there and `except` to return `polys` directly without converting into `geopandas`.

I know it's usually better to be consistent on what we return, but if you have `geopandas` it's a pain not to return it as a `GeoSeries` and drop the functionality entirely for not being able to wrap a list of geometries into `geopandas` is also a bit awkward.

Comment (weikang9009) [reply esda/adbscan.py:507]: Option b) looks more reasonable to me!

Comment (ljwolf) [reply esda/adbscan.py]: I've done this by defining a `GeoSeries` function when the import fails that just converts the input into a list and ignores index/crs input.

**Comment by ljwolf** [comment]
Thank you for the review @weikang9009 @darribas! 

This should now be ready to merge.

**Comment by ljwolf** [comment]
Not sure why the additional commits were introduced. Possibly through an erroneous rebase? Still, should not affect the core of the submission on adbscan.py?

**Comment by sjsrey** [comment]
> Not sure why the additional commits were introduced. Possibly through an erroneous rebase? Still, should not affect the core of the submission on adbscan.py?

This should pass once #124 is merged.

**Pull request merged**